src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php

```diff
@@ -26,17 +26,34 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Gábor Egyed <gabor.egyed@gmail.com>
+ *
+ * @final since version 3.4
  */
 class AssetsInstallCommand extends ContainerAwareCommand
 {
     const METHOD_COPY = 'copy';
     const METHOD_ABSOLUTE_SYMLINK = 'absolute symlink';
     const METHOD_RELATIVE_SYMLINK = 'relative symlink';
 
+    private $filesystem;
+
     /**
-     * @var Filesystem
+     * @param Filesystem $filesystem
      */
-    private $filesystem;
+    public function __construct($filesystem = null)
+    {
+        parent::__construct();
+
+        if (!$filesystem instanceof Filesystem) {
+            @trigger_error(sprintf('Passing a command name as the first argument of "%s" is deprecated since version 3.4 and will be removed in 4.0. If the command was registered by convention, make it a service instead.', __METHOD__), E_USER_DEPRECATED);
+
+            $this->setName(null === $filesystem ? 'assets:install' : $filesystem);
```

Comment:
`parent::__construct($filesystem);`? calling `setName()` now is a behavior change, it's called in `configure` anyway.